axiom/monitors.go

```diff
@@ -67,9 +67,49 @@ func (c *Operator) UnmarshalJSON(b []byte) (err error) {
 	return err
 }
 
+// MonitorType represents the type of the monitor.
+type MonitorType uint8
+
+// All available [Monitor] [Types]s.
+const (
+	MonitorTypeThreshold  MonitorType = iota // Threshold
+	MonitorTypeMatchEvent                    // MatchEvent
+)
+
+func typeFromString(s string) (c MonitorType) {
+	switch s {
+	case MonitorTypeMatchEvent.String():
+		return MonitorTypeMatchEvent
+	default:
+		return MonitorTypeThreshold
```

Comment:
Keep in mind that this has the side effect of defaulting to `Threshold` if the API starts returning anything that's not `MatchEvent`. E.g. if you introduce a new type.

Not a blocker, tho.

Reply:
It's a fair point, this assumption does hold for now, but it won't in a months time. However, I don't want this to error when we introduce a new type. I think let's leave it for now, and I'll update it simultaneously when we do have a new type (as there will be new properties too).